core/shell.py

```diff
@@ -684,6 +684,13 @@ def Main(lang, arg_r, environ, login_shell, loader, readline):
     func_init.SetGlobalFunc(mem, '_end', func_eggex.End(mem))
     func_init.SetGlobalFunc(mem, 'join', func_misc.Join())
     func_init.SetGlobalFunc(mem, 'maybe', func_misc.Maybe())
+    func_init.SetGlobalFunc(mem, 'type', func_misc._Type())
```

Comment:
Generally speaking I use a leading `_` to mean "private"

And a trailing `_` to avoid name conflicts, for example `word_.py`, or `default_` to avoid the  C++ keyword



So I think we could have `List_` and `Dict_` ?   Not sure if we need to avoid name conflicts on the others

Reply:
Ah, yeah. There were conflicts on the `value.*` names when compiling C++, so I just treated all of these to the same renaming. `Type` almost certainly didn't need it. I'm surprised namespacing didn't help prevent the others... I'll look a little deeper into that and switch to trailing `_` if there isn't some other way around this